src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,37 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_ALLOWED_AGE = 18;
+    private static final int MINIMUM_ALLOWED_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RuntimeException("User can't bee null!");
+        }
+        if (user.getLogin() == null) {
+            throw new RuntimeException("Login can't bee null!");
+        }
+        if (user.getLogin().isBlank()) {
+            throw new RuntimeException("Login can't bee empty");
+        }
+        if (user.getPassword() == null) {
+            throw new RuntimeException("Password can't bee null!");
+        }
+        if (user.getPassword().isBlank()) {
+            throw new RuntimeException("Password can't bee empty!");
+        }
+        if (user.getPassword() != null
+                && user.getPassword().length() < MINIMUM_ALLOWED_PASSWORD_LENGTH) {
```

Comment:
```suggestion
        if (user.getPassword().length() < MINIMUM_ALLOWED_PASSWORD_LENGTH) {
```
" user.getPassword() != null " already checked 2 step up